### ISSUE TITLE
Support SQL language injection in clj source files

### ISSUE THREAD

**Issue by JonyEpsilon** [cursive-ide/cursive#196]
IntelliJ's SQL language injection is something I find really useful. It doesn't seem to work in clj files!

**Comment by CmdrDats** [comment]
A starting point that I might suggest is to just enable custom defined language injection (preferences -> Editor -> Language Injections) to start with and let users define their own injection patterns. Over time, you could bundle the most useful injections, but that's less important that just at least having the feature.

In my specific use case, I have the jdbc stuff abstracted (so no hard DELETE, UPDATE or INSERT) - but I use a (prepare-query "SELECT ...")  function for the querying, so I'd want just that to match and inject on that basis.

**Comment by alexanderkiel** [comment]
In my opinion everybody who needs to use SQL should have a look at yesql. With Yesql, you put the SQL strings in an extra file were IntelliJ can use all its SQL magic.

It's an anti pattern anyway to use strings for things were using data would be better. Sadly there will be never a DSL which suits all SQL dialects.

**Comment by CmdrDats** [comment]
@alexanderkiel use of yesql absolutely obviates the need for this, and yesql is something I've considered when I built our system, but in our case it makes a lot more sense to have the SQL right next to the functions that call them (they are highly localised).  

Each to their own opinion - all I'm saying is that I would really appreciate the injection feature in Cursive.

**Comment by cursiveclojure** [comment]
Yeah, I would like to add this. Unfortunately adding a pattern language for the language injection is not trivial. I'll try to do this soon.

**Comment by bilekt3** [comment]
Just saw IntelliJ IDEA Tips & Tricks and played about hour with inject languages and this is things which I really miss in Cursive.

**Comment by DanEEStar** [comment]
@cursive-ide Just wanted to ask if this feature is still on the roadmap?
Just today I thought it would be really useful, also just the SQL keyword route.
